server/src/__test__/users.test.ts

```diff
@@ -2,6 +2,73 @@
 // eslint-disable-next-line import/no-extraneous-dependencies
 import supertest from 'supertest';
 import app from '../app';
+import sync from '../database/sync';
+
+beforeAll(() => sync());
+
+describe('POST /api/v1/signup', () => {
+  test('success sign up', (done) => {
+    supertest(app)
+      .post('/api/v1/signup')
+      .send({
+        name: 'yosra',
+        password: 'password',
+        confirmPassword: 'password',
+        email: 'yosra@gmail.com',
+        address: 'gaza',
+        mobile: '0594121212',
+      })
+      .expect(201)
+      .end((err, res) => {
+        if (err) return done(err);
+        const token = res.header['set-cookie'][0].split('=')[0];
+        expect(token).toBe('token');
+        expect(res.body.success).toBe(true);
+        return done();
+      });
+  });
+});
+describe('validation error', () => {
+  test('validation error, must be a valid email', (done) => {
+    supertest(app)
+      .post('/api/v1/signup')
+      .send({
+        name: 'yosra',
+        password: 'password',
+        confirmPassword: 'password',
+        email: 'yosragmail.com',
+        address: 'gaza',
+        mobile: '0594121212',
+      })
+      .expect(200)
```

Comment:
in validation errors the status should not be `200` you can use `400` or  `422`
i prefer the `422` status code for validation errors